src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setLocalDateTime(LocalDateTime.now());
+        order.setTickets(new ArrayList<>(shoppingCart.getTickets()));
+        order.setUser(shoppingCart.getUser());
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        return orderDao.add(order);

```

Comment:
як на мене буде логічніше очистити шопінг карту вже після того як ми зберегли Order, просто на етапі збереження у нас можуть статись помилки і транзакція не буде виконана, тоді ми і Order не сформували і квитки з shoppingCart вже видалили
```suggestion
        orderDao.add(order);
        shoppingCartService.clearShoppingCart(shoppingCart);
        return order;

```